storage/src/main/java/tech/pegasys/teku/storage/protoarray/ProtoArray.java

```diff
@@ -622,24 +622,11 @@ public boolean nodeIsViableForHead(ProtoNode node) {
       return false;
     }
 
-    // The voting source should be at the same height as the store's justified checkpoint
+    // The voting source should be either at the same height as the store's justified checkpoint or
+    // not more than two epochs ago
     boolean correctJustified =
-        doesCheckpointEpochMatch(node.getJustifiedCheckpoint(), justifiedCheckpoint);
-
-    // If this is a pulled-up block from the current epoch, also check that the unrealized
-    // justification is higher than the store's justified checkpoint, and the voting source is not
-    // more than two epochs ago.
-    if (!correctJustified
-        && !node.getUnrealizedJustifiedCheckpoint().equals(node.getJustifiedCheckpoint())) {
-      correctJustified =
-          node.getUnrealizedJustifiedCheckpoint()
-                  .getEpoch()
-                  .isGreaterThanOrEqualTo(justifiedCheckpoint.getEpoch())
-              && node.getJustifiedCheckpoint()
-                  .getEpoch()
-                  .plus(2)
-                  .isGreaterThanOrEqualTo(currentEpoch);
-    }
+        doesCheckpointEpochMatchOrIsRecentEnough(
+            node.getJustifiedCheckpoint().getEpoch(), justifiedCheckpoint.getEpoch());
```

Comment:
if this evaluates to false we could technically skip the rest of the function, because we've already failed the final check....